src/daemon/backend.cpp

```diff
@@ -0,0 +1,165 @@
+// Copyright (C) 2019 Luxoft Sweden AB
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+//
+// SPDX-License-Identifier: MPL-2.0
+
+#include "daemon/backend.h"
+
+#include <algorithm>
+#include <cassert>
+#include <utility>
+
+namespace ConnectivityManager::Daemon
+{
+    Backend::Backend() = default;
+
+    Backend::~Backend() = default;
+
+    void Backend::critical_error()
+    {
+        signals_.critical_error.emit();
+    }
+
+    void Backend::wifi_status_set(WiFiStatus status)
+    {
+        if (state_.wifi.status == status)
+            return;
+
+        state_.wifi.status = status;
+        signals_.wifi.status_changed.emit(status);
+    }
+
+    Backend::WiFiAccessPoint::Id Backend::wifi_access_point_next_id()
+    {
+        while (true) {
+            wifi_access_point_last_id_++;
+
+            if (wifi_access_point_last_id_ == WiFiAccessPoint::ID_EMPTY)
+                wifi_access_point_last_id_++;
+
+            if (!wifi_access_point_find(wifi_access_point_last_id_))
+                break;
+        }
+
+        return wifi_access_point_last_id_;
+    }
+
+    Backend::WiFiAccessPoint *Backend::wifi_access_point_find(WiFiAccessPoint::Id id)
+    {
+        auto i = state_.wifi.access_points.find(id);
+
+        return i == state_.wifi.access_points.cend() ? nullptr : &i->second;
+    }
+
+    void Backend::wifi_access_points_add_all(std::vector<WiFiAccessPoint> &&access_points)
+    {
+        wifi_access_points_remove_all();
+
+        for (WiFiAccessPoint &access_point : access_points) {
+            WiFiAccessPoint::Id id = access_point.id;
+            assert(id != WiFiAccessPoint::ID_EMPTY);
+
+            state_.wifi.access_points.emplace(id, std::move(access_point));
+        }
+
+        signals_.wifi.access_points_changed.emit(WiFiAccessPoint::Event::ADDED_ALL, nullptr);
+    }
+
+    void Backend::wifi_access_points_remove_all()
+    {
+        if (state_.wifi.access_points.empty())
+            return;
+
+        state_.wifi.access_points.clear();
+
+        signals_.wifi.access_points_changed.emit(WiFiAccessPoint::Event::REMOVED_ALL, nullptr);
+    }
+
+    void Backend::wifi_access_point_add(WiFiAccessPoint &&access_point)
+    {
+        WiFiAccessPoint::Id id = access_point.id;
+        assert(id != WiFiAccessPoint::ID_EMPTY);
+
+        auto result = state_.wifi.access_points.emplace(id, std::move(access_point));
+
+        signals_.wifi.access_points_changed.emit(WiFiAccessPoint::Event::ADDED_ONE,
+                                                 &result.first->second);
+    }
+
+    void Backend::wifi_access_point_remove(const WiFiAccessPoint &access_point)
+    {
+        auto i = state_.wifi.access_points.find(access_point.id);
+        if (i == state_.wifi.access_points.cend()) {
+            assert(false);
```

Comment:
I do not know. I think I would rather just remove the `assert()`. Will see how the meson.bbclass works by default. If it does not make it so `-DNDEBUG` is passed in release builds we should probably not user assert() too much since it should be used to indicate "programming error"... but lead to unexpected crashes in production systems.

Reply:
Turns out poky does not add -DNDEBUG by default. Removed this assert since it is not critical. Resolving.